### PR TITLE
code coverage

### LINKED ISSUE #11: code coverage for juttle-viz
add code coverage checking for juttle-viz so we can figure out where the testing efforts should be focused.

### LINKED ISSUE #16: move phantom based tests to node only tests
The idea here is to not depend on phantom to test and simplify the way tests are written and executed here.

### PR DESCRIPTION
**Pull request by rlgomes**
fixes #11
fixes #16

now we can see where the actual gaps in tests reside

### CODE REVIEW

**Comment by rlgomes** [comment]
@go-oleg @mnibecker @demmer @dmehra  a little bit of magic with `jsdom` to make things work in node but at least this way we don't depend on using a browser to test and we can get real code coverage numbers. They're not fantastic but here are the details:

**Summary:**

```
=============================== Coverage summary ===============================
Statements   : 56.55% ( 2894/5118 )
Branches     : 45.06% ( 889/1973 )
Functions    : 51.23% ( 521/1017 )
Lines        : 57.01% ( 2798/4908 )
================================================================================
```

**Full Report:**

https://travis-ci.org/juttle/juttle-viz/builds/102706962#L800

At least now we know where to put our efforts to get things done.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
-node_modules
 build
-/lib
+coverage
+lib
```
Comment (go-oleg) [inline .gitignore]: The `/lib` is necessary to target only the `lib` that is generated by the gulp build task (and not the one under `src`). Filed https://github.com/juttle/juttle-viz/issues/22 to address the issue of multiple `lib` folders.

File: test/fakebrowser.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            element._hacked = true;
+            return element;
+        };
+
+        // exposed globally for tests under lib/generators
+        global.HTMLElement = window.HTMLElement;
+
+        var d3 = require('d3');
+        // d3 transformations will simply be mocked out this way
+        d3.transform = function() {
```
Comment (go-oleg) [inline test/fakebrowser.js]: Out of curiosity, why does this need to be mocked?

File: gulpfile.js [hunk 3]
```diff
@@ -56,54 +56,57 @@ gulp.task('example-serve', ['browserify', 'styles', 'watch'], function() {
...
-        'test/lib/**/*.spec.js',
-
-        // exclude lib browser tests
-        '!test/lib/charts/*.spec.js',
-        '!test/lib/generators/*.spec.js',
-        '!test/lib/components/*.spec.js'
+gulp.task('instrument', function () {
+    return gulp.src([
+        'src/**/*.js',
+        '!src/lib/components/log-explorer/**/*.js'
```
Comment (go-oleg) [inline gulpfile.js]: I'm guessing this is here cause isparta or someone else isn't happy with jsx? This prompted me to wonder if its more correct to run tests against the original code or transpiled code in `lib` (a la [this stackoverflow question](http://stackoverflow.com/questions/33228754/in-a-babel-project-workflow-does-it-make-more-sense-to-run-unit-tests-firstly-a)). From a quick look at two libraries I've been using recently, [react-select](https://github.com/JedWatson/react-select) and [redux](https://github.com/rackt/redux), both test against the original code. Don't have a strong preference either way.

**Comment by go-oleg** [comment]
Some comments/questions above, otherwise looks good!

File: test/fakebrowser.js (hunk 2 above)
Comment (rlgomes) [inline test/fakebrowser.js]: jsdom doens't support any CSS attribute setting doesn't really happen underneath the curtains which is part of the DOM setting transform values for scale/translate etc. By simply defining these here I made the calls have no effect on the run.

File: gulpfile.js (hunk 3 above)
Comment (rlgomes) [inline gulpfile.js]: let me check as I think the ispart thing isn't necessary and I may have **copy & pasta'd** a little too much.

File: .gitignore (hunk 1 above)
Comment (rlgomes) [inline .gitignore]: fixing...

**Comment by rlgomes** [comment]
@go-oleg few tweaks as you suggested and I also fixed up the issue with log-explorer which did need the isparta instrumenter but also required the babelrc file to indicate that we needed the react plugin when parsing/instrumenting the code.

**Comment by go-oleg** [comment]
looks good

**Pull request merged**